slight/src/commands/run.rs

```diff
@@ -147,132 +147,87 @@ fn build_store_instance(
             bail!("unsupported toml spec version");
         };
 
+        if resource_type != "events" && resource_type != "http" {
+            maybe_add_named_capability_to_store(
+                &toml.specversion,
+                toml.secret_store.clone(),
+                &mut capability_store,
+                c.clone(),
+                resource_map.clone(),
+                &toml_file_path,
+            )?;
+        }
+
         match resource_type {
             "events" => {
                 if !linked_capabilities.contains("events") {
-                    builder.link_capability::<Events<Builder>>(resource_type.to_string())?;
+                    let events = Events::<Builder>::new(resource_map.clone());
+                    builder
+                        .link_capability::<Events<Builder>>()?
+                        .add_to_builder("events".to_string(), events);
                     linked_capabilities.insert("events".to_string());
-
-                    slight_builder =
-                        slight_builder.add_state(State::Events(EventsState::<Builder>::new(
-                            resource_map.clone(),
-                        )))?;
                 } else {
                     bail!("the events capability was already linked");
                 }
             }
             _ if KV_HOST_IMPLEMENTORS.contains(&resource_type) => {
                 if !linked_capabilities.contains("kv") {
-                    builder.link_capability::<Kv>("kv".to_string())?;
+                    builder.link_capability::<Kv>()?;
                     linked_capabilities.insert("kv".to_string());
                 }
 
-                maybe_add_named_capability_to_store(
-                    &toml.specversion,
-                    toml.secret_store.clone(),
-                    &mut capability_store,
-                    c.clone(),
-                    resource_map.clone(),
-                    &toml_file_path,
-                )?;
-
-                slight_builder = slight_builder.add_state(State::Kv(slight_kv::KvState::new(
-                    resource_type.to_string(),
-                    capability_store.clone(),
-                )))?;
+                let resource =
+                    slight_kv::Kv::new(resource_type.to_string(), capability_store.clone());
+                builder.add_to_builder("kv".to_string(), resource);
```

Comment:
The previous alternative to this (i.e., `maybe_add_named_capability_to_store`) was performing some error handling - like, bailing when two capabilities of the same name are added, and whatnot. Is any of that present now?

Reply:
I believe so. I moved that function invocation to above https://github.com/deislabs/spiderlightning/pull/240/files#diff-81fdd571c3ec358ab5f1a5984fd30948d9db1ecfa8d5ff2f016e5a412a9f7d35R146-R153